cdk/ecs/lib/stacks/ecsStack.ts

```diff
@@ -0,0 +1,1118 @@
+                CONFIG_SERVER_URL: `http://${this.CONFIG_SERVER}-DNS.${this.serviceDiscoveryStack.namespace.namespaceName}:8888`,
+                DISCOVERY_SERVER_URL: `http://${this.DISCOVERY_SERVER}-DNS.${this.serviceDiscoveryStack.namespace.namespaceName}:8761/eureka`,
+                VISITS_SERVICE_IP: `${this.VISITS_SERVICE}-DNS.${this.serviceDiscoveryStack.namespace.namespaceName}`,
+            },
+            logging: LogDrivers.awsLogs({
+                streamPrefix: 'ecs',
+                logGroup: visitsLogGroup,
+            }),
+        });
+
+        // Add Port Mapping
+        mainContainer.addPortMappings({
+            containerPort: 8082,
+            protocol: Protocol.TCP,
+        });
+
+        mainContainer.addMountPoints({
+            sourceVolume: 'opentelemetry-auto-instrumentation',
+            containerPath: '/otel-auto-instrumentation',
+            readOnly: false,
+        });
+
+        // Add init container
+        const initContainer = taskDefinition.addContainer('init-visits-service-container', {
+            image: ContainerImage.fromRegistry(
+                `public.ecr.aws/aws-observability/adot-autoinstrumentation-java:${adotJavaImageTag}`,
+            ),
+            essential: false, // The container will stop with exit 0 after it completes.
+            command: ['cp', '/javaagent.jar', '/otel-auto-instrumentation/javaagent.jar'],
+        });
+
+        initContainer.addMountPoints({
+            sourceVolume: 'opentelemetry-auto-instrumentation',
+            containerPath: '/otel-auto-instrumentation',
+            readOnly: false,
+        });
+
+        // Add CloudWatch agent container
+        taskDefinition.addContainer('ecs-cwagent-visits-service-container', {
+            image: ContainerImage.fromRegistry('public.ecr.aws/cloudwatch-agent/cloudwatch-agent:latest'),
+            memoryLimitMiB: 128,
+            essential: true,
+            environment: {
+                CW_CONFIG_CONTENT: JSON.stringify({
+                    traces: {
+                        traces_collected: {
+                            application_signals: {},
+                        },
+                    },
+                    logs: {
+                        metrics_collected: {
+                            application_signals: {
+                                rules: [this.DISCOVERY_SERVER_CW_CONFIG, this.CONFIG_SERVER_CW_CONFIG],
+                            },
+                        },
+                    },
+                }),
+            },
+
+            logging: LogDrivers.awsLogs({
+                streamPrefix: 'ecs',
+                logGroup: cwAgentVisitsServiceLogGroup,
+            }),
+        });
+
+        // Create ECS service
+        this.createService({
+            serviceName: this.VISITS_SERVICE,
+            taskDefinition: taskDefinition,
+        });
+    }
+
+    runInsuranceService(adotPythonImageTag: string, dbSecret: SmSecret, rds_endpoint: string) {
```

Comment:
it's better to extract some common code out from `runInsuranceService` and `runBillingService`